api/tests/provider_api_tests.py

```diff
@@ -52,3 +53,36 @@ def test_get_list_of_providers(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(list(response.data[0].values())[1], self.provider1.person.name)
         self.assertEqual(len(response.data), 3)
+
+
+class ProviderFilteringTestCase(BaseAPITest):
+    def setUp(self):
+        super().setUp()
+        self.provider1 = ProviderFactory.create()
+        self.provider2 = ProviderFactory.create()
+        self.provider3 = ProviderFactory.create()
+
+    def test_when_parameters_are_correct(self):
+        request = self.request_factory.get(
+            "/api/providers/", {"average_rating_min": 0, "average_rating_max": 5}
+        )
```

Comment:
You defined the `get_list_response` in the `BaseAPITest` which can be adapted in order to be used  here